src/openrct2/actions/RideSetColourScheme.hpp

```diff
@@ -0,0 +1,71 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2018 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../Cheats.h"
+#include "../common.h"
+#include "../core/MemoryStream.h"
+#include "../interface/Window.h"
+#include "../localisation/Localisation.h"
+#include "../localisation/StringIds.h"
+#include "../management/Finance.h"
+#include "../ride/Ride.h"
+#include "../world/Park.h"
+#include "../world/Sprite.h"
+#include "GameAction.h"
+
+struct RideSetColourSchemeAction : public GameActionBase<GAME_COMMAND_SET_COLOUR_SCHEME, GameActionResult>
+{
+private:
+    int32_t _x = 0, _y = 0, _z = 0, _direction = 0, _trackType = 0;
+    uint16_t _newColourScheme = 0;
+
+public:
+    RideSetColourSchemeAction() = default;
+    RideSetColourSchemeAction(int32_t x, int32_t y, int32_t z, int32_t direction, int32_t trackType, uint16_t newColourScheme)
+        : _x(x)
+        , _y(y)
+        , _z(z)
+        , _direction(direction)
+        , _trackType(trackType)
+        , _newColourScheme(newColourScheme)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags() | GA_FLAGS::ALLOW_WHILE_PAUSED;
+    }
+
+    void Serialise(DataSerialiser& stream) override
+    {
+        GameAction::Serialise(stream);
+
+        stream << DS_TAG(_x) << DS_TAG(_y) << DS_TAG(_z) << DS_TAG(_direction) << DS_TAG(_trackType)
+               << DS_TAG(_newColourScheme);
+    }
+
+    GameActionResult::Ptr Query() const override
+    {
+        return std::make_unique<GameActionResult>();
+    }
+
+    GameActionResult::Ptr Execute() const override
+    {
+        GameActionResult::Ptr res = std::make_unique<GameActionResult>();
+        res->ExpenditureType = RCT_EXPENDITURE_TYPE_RIDE_CONSTRUCTION;
+        res->ErrorTitle = STR_CANT_SET_COLOUR_SCHEME;
```

Comment:
You set the error title here, but is this error occurring at any point?

Reply:
No. Or not yet, anyway. I figured it might come in handy if more checks need to be implemented later.

Reply:
Remove it then? Provide a comment? Either way, I'm not very fussed about it.